tool/tbot/main.go

```diff
@@ -66,126 +60,92 @@ access Teleport protected resources in the same way your engineers do!
-	kubeCredentialsCmd.Flag("destination-dir", "The destination directory with which to generate Kubernetes credentials").Required().StringVar(&cf.DestinationDir)
+	// TODO: consider discarding config flag for non-legacy. These should always be self contained.
+
+	// Initialize all new-style commands.
+	var commands []cli.CommandRunner
+	commands = append(commands,
+		cli.NewInitCommand(app, func(init *cli.InitCommand) error {
+			return onInit(globalCfg, init)
+		}),
+
+		cli.NewMigrateCommand(app, func(migrateCfg *cli.MigrateCommand) error {
+			return onMigrate(ctx, globalCfg, migrateCfg, stdout)
+		}),
+
+		cli.NewSSHProxyCommand(app, func(sshProxyCommand *cli.SSHProxyCommand) error {
+			return onSSHProxyCommand(ctx, globalCfg, sshProxyCommand)
+		}),
+
+		cli.NewProxyCommand(app, func(proxyCmd *cli.ProxyCommand) error {
+			return onProxyCommand(ctx, globalCfg, proxyCmd)
+		}),
+
+		cli.NewDBCommand(app, func(dbCmd *cli.DBCommand) error {
+			return onDBCommand(globalCfg, dbCmd)
+		}),
+
+		cli.NewSSHMultiplexerProxyCommand(app, func(c *cli.SSHMultiplerProxyCommand) error {
+			return onSSHMultiplexProxyCommand(ctx, c.Socket, c.Data)
+		}),
 
+		cli.NewKubeCredentialsCommand(kubeCmd, func(kubeCredentialsCmd *cli.KubeCredentialsCommand) error {
+			return onKubeCredentialsCommand(ctx, globalCfg, kubeCredentialsCmd)
+		}),
+
+		// `start` and `configure` commands
+		cli.NewLegacyCommand(startCmd, buildConfigAndStart(ctx, globalCfg)),
+		cli.NewLegacyCommand(configureCmd, buildConfigAndConfigure(ctx, globalCfg, &configureOutPath, stdout)),
+
+		cli.NewIdentityCommand(startCmd, buildConfigAndStart(ctx, globalCfg)),
+		cli.NewIdentityCommand(configureCmd, buildConfigAndConfigure(ctx, globalCfg, &configureOutPath, stdout)),
+
+		cli.NewDatabaseCommand(startCmd, buildConfigAndStart(ctx, globalCfg)),
+		cli.NewDatabaseCommand(configureCmd, buildConfigAndConfigure(ctx, globalCfg, &configureOutPath, stdout)),
+
+		cli.NewKubernetesCommand(startCmd, buildConfigAndStart(ctx, globalCfg)),
+		cli.NewKubernetesCommand(configureCmd, buildConfigAndConfigure(ctx, globalCfg, &configureOutPath, stdout)),
+
+		cli.NewApplicationCommand(startCmd, buildConfigAndStart(ctx, globalCfg)),
+		cli.NewApplicationCommand(configureCmd, buildConfigAndConfigure(ctx, globalCfg, &configureOutPath, stdout)),
+
+		cli.NewApplicationTunnelCommand(startCmd, buildConfigAndStart(ctx, globalCfg)),
+		cli.NewApplicationTunnelCommand(configureCmd, buildConfigAndConfigure(ctx, globalCfg, &configureOutPath, stdout)),
+
+		cli.NewDatabaseTunnelCommand(startCmd, buildConfigAndStart(ctx, globalCfg)),
+		cli.NewDatabaseTunnelCommand(configureCmd, buildConfigAndConfigure(ctx, globalCfg, &configureOutPath, stdout)),
+
+		cli.NewSPIFFEX509SVIDCommand(startCmd, buildConfigAndStart(ctx, globalCfg)),
+		cli.NewSPIFFEX509SVIDCommand(configureCmd, buildConfigAndConfigure(ctx, globalCfg, &configureOutPath, stdout)),
+	)
+
+	// TODO: workload id / spiffe service subcommand?
```

Comment:
Let's leave this out for now I think - the framework you've added here makes it easy enough for me to add this once I've identified the simplest possible use-case.